gammapy/datasets/map.py

```diff
@@ -169,13 +197,121 @@ def _default_binsz(observation, spatial_bin_size_min=0.01 * u.deg):
+    energy_axis_true : `~gammapy.maps.MapAxis`, optional
+        True energy axis. Default is None.
+        If None it is determined from the observation IRFs.
+    energy_axis : `~gammapy.maps.MapAxis`, optional
+        Reconstructed energy axis. Default is None.
+        If None it is determined from the observation IRFs.
+    energy_bin_per_decade_max : int, optional
+        Maximal number of bin per decade in energy for the reference dataset
+    spatial_width : `~astropy.units.Quantity`, optional
+        Spatial window size. Default is None.
+         If None it is determined from the observation offset max or rad max.
+    spatial_width_max : `~astropy.quantity.Quantity`, optional
+        Maximal spatial width. Default is 12 degree.
+    spatial_bin_size : `~astropy.units.Quantity`, optional
+        Pixel size. Default is None.
+        If None it is determined from the observation PSF R68.
+    spatial_bin_size_min : `~astropy.quantity.Quantity`, optional
+        Minimal spatial bin size. Default is 0.01 degree.
+    position : `~astropy.coordinates.SkyCoord`, optional
+        Center of the geometry. Defalut is the observation pointing.
+    frame: str, optional
+        frame of the coordinate system. Defalut is icrs.
+    """
+
+    if spatial_width is None:
+        spatial_width = _default_width(data, spatial_width_max)
+    if spatial_bin_size is None:
+        spatial_bin_size = _default_binsz(data, spatial_bin_size_min)
+
+    if energy_axis is None or energy_axis_true is None:
+        energy_axis_, energy_axis_true_ = _default_energy_axis(
+            data, energy_bin_per_decade_max
+        )
+
+        if energy_axis is None:
+            energy_axis = energy_axis_
+
+        if energy_axis_true is None:
+            energy_axis_true = energy_axis_true_
+
+    if dataset_name is None:
+        if hasattr(data, "obs_id"):
+            dataset_name = f"obs_{data.obs_id}"
+        else:
+            dataset_name = f"{data.name}"
+
+    if position is None:
+        if hasattr(data, "pointing"):
+            if data.pointing.mode is not PointingMode.POINTING:
+                raise NotImplementedError(
+                    "Only datas with fixed pointing in ICRS are supported"
+                )
+            position = data.pointing.fixed_icrs
+        else:
+            position = data.exposure.geom.center_skydir
+
+    geom = WcsGeom.create(
+        skydir=position.transform_to(frame),
+        width=spatial_width,
+        binsz=spatial_bin_size.to_value(u.deg),
+        frame=frame,
+        axes=[energy_axis],
+    )
+
+    axes = dict(
+        energy_axis_true=energy_axis_true,
+    )
+    if data.edisp is not None:
+        if hasattr(data.edisp, "axes"):
+            axes["migra_axis"] = data.edisp.axes["migra"]
+        else:
+            axes["migra_axis"] = data.edisp.edisp_map.geom.axes["migra"]
+
+    dataset = MapDataset.create(
+        geom,
+        name=dataset_name,
+        **axes,
```

Comment:
What about the `irf_binsz` ? I think this one is important to cover as well.

Reply:
yes I will have to think how to define good default for that in an other PR.